src/walk-manifest.js

```diff
@@ -129,15 +144,19 @@ var walkPlaylist = function(decrypt, basedir, uri, parent, manifestIndex) {
 			return;
 		}
 		// put segments in manifest-name/segment-name.ts
-		s.file = path.join(path.dirname(manifest.file), path.basename(s.uri));
+		s.file = path.join(path.dirname(manifest.file), baseQueryStringName(s.uri));
 		if (!isAbsolute(s.uri)) {
 			s.uri = joinURI(path.dirname(manifest.uri), s.uri);
 		}
 		if (key) {
 			s.key = key;
 			s.key.iv = s.key.iv || new Uint32Array([0, 0, 0, manifest.parsed.mediaSequence, i]);
 		}
-		manifest.content = new Buffer(manifest.content.toString().replace(s.uri, path.basename(s.uri)));
+
+		manifest.content = new Buffer(manifest.content.toString().replace(
+			s.uri,
+			path.relative(path.dirname(manifest.file), s.file)
```

Comment:
instead of using the uri, we have to use the file here. Its possible that this was an issue.